### DISCUSSION TITLE
Docs on formState within useForm appear to be incorrect?

### DISCUSSION THREAD

**Discussion by proevilz** [react-hook-form/react-hook-form#5572] [Q&A]
[This docs page shows the following](https://react-hook-form.com/api/useform/formstate) on how to subscribe to `formState` via `useEffect`.

```javascript
import React from "react";
import { useForm } from "react-hook-form";

export default function App () {
  const {
    register,
    handleSubmit,
    formState: { touched }
  } = useForm();

  const onSubmit = (data) => console.log(data);

  React.useEffect(() => {
    console.log("touched", formState.touched);
  },[formState]); // use entire formState object as optional array arg in useEffect, not individual properties of it


  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("test")} />
      <input type="submit" />
    </form>
  );
};
```

But this just throws an error on the `useEffect` - this is because we destructure `touched` instead of pulling in `formState` 
> ReferenceError: formState is not defined

Sandbox: https://codesandbox.io/s/green-fog-5owm9?file=/src/App.js

Is it just me, or are the docs wrong on this?

**Comment by bluebill1049** [comment]
we will need `formState` as dep, because we are batching form state update with `formState`. This is probably something can be improved with React 18 batching update.
